maven-plugin/src/main/java/org/openmrs/maven/plugins/BuildDistro.java

```diff
@@ -587,4 +592,85 @@ public boolean accept(File dir, String name) {
 			}
 		}
 	}
+
+	/**
+	 * Reads the content.properties file.
+	 * <p> For dependencies defined in both content.properties and distro.properties, this ensures the version specified in distro.properties 
+	 * matches the range defined in content.properties; otherwise, we output an error to the user to fix this.</p>
+	 * For dependencies not defined in distro.properties but defined in content.properties, the SDK should locate the latest matching version and add that to the OMODs and ESMs used by the distro.
+	 * 
+	 * @param userDir
+	 * @param distroProperties
+	 * @throws MojoExecutionException
+	 */
+	private void parseContentProperties(File userDir, DistroProperties distroProperties) throws MojoExecutionException {
```

Comment:
These checks should be implemented for both the `build-distro` and `setup` commands, so it might make sense to extract the "meat" of them into one of the shared components (or add a new one).